### PR TITLE
Update uiSref input type to match functionality

### PR DESCRIPTION
**Pull request by oBusk**
Was updating our code to try to use strict templates and was noting failures on templates that has been working for years. Noticed that the inputtype to `uiSref` is "string" when it works just fine with a declaration or object too, since the internal uses all support StateOrName.

### CODE REVIEW

**Review by christopherthielen — APPROVED** [review]

File: src/directives/uiSref.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class UISref implements OnChanges {
    * <a uiSref="hoome">Home</a>
    * ```
    */
-  @Input('uiSref') state: string;
+  @Input('uiSref') state: StateOrName;
```
Comment (christopherthielen) [inline src/directives/uiSref.ts:90]: Should this be changed on line 125 too?

Comment (oBusk) [reply src/directives/uiSref.ts:90]: I'm confused? L125 in this file?

https://github.com/ui-router/angular/blob/a34d95a3994bc207aa3365b1c666486f72ec748c/src/directives/uiSref.ts#L125

Comment (christopherthielen) [reply src/directives/uiSref.ts:90]: oops, typo.  I meant Line 135

```
  /** @internal */
  set uiSref(val: string) { // here
    this.state = val;
    this.update();
  }
```

Comment (christopherthielen) [reply src/directives/uiSref.ts:90]: I do believe the setter is how the data-binding is applied 
```
  @Input('uiSref') state: StateOrName;
```

Comment (oBusk) [reply src/directives/uiSref.ts:90]: Hmm well the decorator should bind the directive input "uiSref" to the class property "state". But nevertheless, whatever it's use, the setter should have the same type as the value it sets, I've updated it.

**Pull request merged**